lib/astarte_device.ex

```diff
@@ -363,25 +366,28 @@ defmodule Astarte.Device do
     handle_disconnected_publish(from)
   end
 
-  def no_certificate(:internal, :request_certificate, data) do
+  def no_certificate(:internal, {:request_certificate, retry}, data) do
```

Comment:
I'd call it `current_attempt`/`attempts` instead of `retry` since with the latter I expect the variable to be a boolean which tells me wether to retry or not.